core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -0,0 +1,149 @@
+   */
+  def isValidComment(comment: String): Boolean = {
+    recommendationsToExclude.forall(excluded => !comment.contains(excluded))
+  }
+
+  def getOperatorScoreFile: String = {
+    s"operatorsScore-$platformType.csv"
+  }
+}
+
+class DatabricksPlatform(platformType: String) extends Platform(platformType) {
+  override val recommendationsToExclude: Seq[String] = Seq(
+    "spark.executor.cores",
+    "spark.executor.instances",
+    "spark.executor.memory",
+    "spark.executor.memoryOverhead"
+  )
+  override val recommendationsToInclude: Seq[(String, String)] = Seq(
+    ("spark.databricks.optimizer.dynamicFilePruning", "false")
+  )
+}
+
+class DataprocPlatform(platformType: String) extends Platform(platformType)
+
+class EmrPlatform(platformType: String) extends Platform(platformType)
+
+class OnPremPlatform extends Platform(PlatformTypes.ONPREM)
+
+/**
+ * Factory for creating instances of different platforms.
+ * This factory supports various platforms and provides methods for creating
+ * corresponding platform instances.
+ */
+object PlatformFactory extends Logging {
+  private lazy val platformInstancesMap: Map[String, Platform] = Map(
+    PlatformTypes.DATABRICKS_AWS -> new DatabricksPlatform(PlatformTypes.DATABRICKS_AWS),
+    PlatformTypes.DATABRICKS_AZURE -> new DatabricksPlatform(PlatformTypes.DATABRICKS_AZURE),
+    // if no GPU specified, then default to dataproc-t4 for backward compatibility
+    PlatformTypes.DATAPROC -> new DataprocPlatform(PlatformTypes.DATAPROC_T4),
+    PlatformTypes.DATAPROC_T4 -> new DataprocPlatform(PlatformTypes.DATAPROC_T4),
+    PlatformTypes.DATAPROC_L4 -> new DataprocPlatform(PlatformTypes.DATAPROC_L4),
+    PlatformTypes.DATAPROC_SL_L4 -> new DataprocPlatform(PlatformTypes.DATAPROC_SL_L4),
+    PlatformTypes.DATAPROC_GKE_L4 -> new DataprocPlatform(PlatformTypes.DATAPROC_GKE_L4),
+    PlatformTypes.DATAPROC_GKE_T4 -> new DataprocPlatform(PlatformTypes.DATAPROC_GKE_T4),
+    // if no GPU specified, then default to emr-t4 for backward compatibility
+    PlatformTypes.EMR -> new EmrPlatform(PlatformTypes.EMR_T4),
+    PlatformTypes.EMR_T4 -> new EmrPlatform(PlatformTypes.EMR_T4),
+    PlatformTypes.EMR_A10 -> new EmrPlatform(PlatformTypes.EMR_A10),
+    PlatformTypes.ONPREM -> new OnPremPlatform
+  )
+
+  /**
+   * Creates an instance of a platform based on the specified platform key.
+   *
+   * @param platformKey The key representing the desired platform.
+   * @return An instance of the specified platform.
+   * @throws IllegalArgumentException if the specified platform key is not supported.
+   */
+  def createInstance(platformKey: String): Platform = {
+    val platformToUse = if (platformKey.isEmpty) {
+      logInfo(s"Platform is not specified, defaulting to ${PlatformTypes.ONPREM}")
+      PlatformTypes.ONPREM
+    } else platformKey
+
+    platformInstancesMap.getOrElse(platformToUse,
+      throw new IllegalArgumentException(s"Platform $platformToUse is not supported"))
```

Comment:
it would be nice to print the supported ones out.

Reply:
Updated message to include supported ones.